mfai/pytorch/models/llms/cross_attention.py

```diff
@@ -62,29 +80,75 @@ def __init__(
             vocab_size=vocab_size,
         )
 
-        # Initialize our resnet50 vision encoder
-        num_channels_viz = (
-            settings.vision_input_shape[2] * settings.vision_input_shape[3]
-        )
-        self.vision_encoder = ResNet50MLM(
-            num_channels=num_channels_viz,
-            num_classes=settings.emb_dim,
-            settings=ResNet50MLMSettings(num_tokens=settings.num_tokens_vision),
-        )
+        self.vision_encoder: WeatherProjector | ResNet50MLM | VitEncoder
+
+        if self.settings.vision_encoder == "linear":
+            input_dims = (
+                settings.vision_input_shape[0],
+                settings.vision_input_shape[1],
+                settings.vision_input_shape[-1],
+            )
+            s = WeatherProjectorSettings(
+                input_dims=input_dims,
+                embedding_dim=self.settings.emb_dim,
+                patch_size=self.settings.patch_size,
+            )
+            self.vision_encoder = WeatherProjector(settings=s)
+
+        elif self.settings.vision_encoder == "resnet50":
+            self.vision_encoder = ResNet50MLM(
+                num_channels=settings.vision_input_shape[3],
+                num_classes=settings.emb_dim,
+                settings=ResNet50MLMSettings(
+                    num_tokens=settings.resnet_num_tokens,
+                    pos_embedding=settings.resnet_pos_embedding,
+                    mlp_output=settings.resnet_mlp_output,
+                ),
+            )
+        elif self.settings.vision_encoder == "vit":
+            # Initialize the ViT encoder, we have one input channel per feature per timestep
+            self.vision_encoder = VitEncoder(
+                in_channels=settings.vision_input_shape[-1],
+                out_channels=settings.emb_dim,
+                settings=ViTEncoderSettings(
+                    emb_dim=settings.emb_dim,
+                    transformer_dropout=settings.drop_rate,
+                    emb_dropout=settings.drop_rate,
+                    autopad_enabled=True,
+                    patch_size=self.settings.patch_size,
+                ),
+                input_shape=settings.vision_input_shape[:2],
+            )
+
+        else:
+            raise ValueError(
+                f"Unknown vision encoder: {self.settings.vision_encoder}. Use 'linear', 'vit' or 'resnet50'."
+            )
+
+        if settings.layer_norm_vis:
+            self.norm_or_ident: nn.Identity | nn.LayerNorm = nn.LayerNorm(
+                self.settings.emb_dim
+            )
+        else:
+            self.norm_or_ident = nn.Identity()
```

Comment:
Type hinting before the conditional definition of the `norm_or_ident` parameter.
```suggestion
        self.norm_or_ident: nn.Identity | nn.LayerNorm
        if settings.layer_norm_vis:
            self.norm_or_ident = nn.LayerNorm(
                self.settings.emb_dim
            )
        else:
            self.norm_or_ident = nn.Identity()
```